internal/pkg/db/postgres/migration.go

```diff
@@ -0,0 +1,235 @@
+//
+// Copyright (C) 2025 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package postgres
+
+import (
+	"context"
+	"embed"
+	goErrors "errors"
+	"fmt"
+
+	"github.com/blang/semver/v4"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v4/clients/logger"
+)
+
+const (
+	migrationTableName = "schema_migrations"
+
+	idempotentDBScriptsDir = "sql/idempotent"
+	versionMigrationDir    = "sql/versions"
+
+	statusSuccess = "SUCCESS"
+	statusFailure = "FAILURE"
+
+	// initialVersion specifies the version before the service introduced schema_migrations table to handle migration
+	// as we introduce the schema_migration in the release of 4.0, the initial version is set to 3.1.1 Napa release
+	initialVersion = "3.1.1"
```

Comment:
since the postgres is initilized from 4.0.0, is it possible to treat 4.0.0 as the initialVersion?
there was no SQL schema in 3.1.1, so it would look wierd

Reply:
Your concern makes sense to me. The original design of `initialVersion` is to specify the version before the service introduced `schema_migrations` table, so 3.1.1 Napa release would be the one for `initialVersion`.  But we don't have any schema as defined in 3.1.1, it does confuse developers. I've updated both the comments and the value to 4.0.0 in https://github.com/edgexfoundry/edgex-go/pull/5073/commits/30df08c41207572b2f54308f0b551a5f939b7019